src/commercetools/testing/abstract.py

```diff
@@ -118,6 +118,11 @@ def delete_by_key(self, key):
             if obj["key"] == key:
                 return self.objects.pop(obj_id)
 
+    def delete_by_container_and_key(self, container, key):
+        for obj in self.objects.values():
+            if obj["container"] == container and obj["key"] == key:
+                return obj
```

Comment:
You're returning the object instead of deleting it. I'd suggest 
```suggestion
        for obj_id, obj in self.objects.items():
            if obj["container"] == container and obj["key"] == key:
                return self.objects.pop(obj_id)
```
Where similar to the delete function above we scan the collection of objects present on `self` and if we find the object we want to delete, we return the whole collection with the exception of the one we want to delete (by popping it from the dict)